### PR TITLE
LE Audio: Fix several small issues with the audio shell

### LINKED ISSUE #50778: LE Audio: Audio shell: Unicast server cannot execute commands for the default_stream
**Describe the bug**
The `default_stream` is not properly for the unicast server, and thus remains NULL. The means that none of the `bt_audio_stream_*` commands can be executed. 

**To Reproduce**
N/A

**Expected behavior**
The `default_stream` should be set and all the commands callable as the unicast server. 

**Impact**
Minor; just missing features for the unicast server. 

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - Commit SHA or Version used: a5ce3da6ab425d104d73d34f47352ae721c5d216

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Thalley**
Fix a couple of minor things in the audio shell, most notably how some of the functions were wrongly guarded. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/50778

### CODE REVIEW

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by kruithofa — DISMISSED** [review]

File: subsys/bluetooth/shell/audio.c [hunk 1]
```diff
@@ -1204,24 +1188,29 @@ static int cmd_list(const struct shell *sh, size_t argc, char *argv[])
...
-static int cmd_connect(const struct shell *sh, size_t argc, char *argv[])
+static int cmd_release(const struct shell *sh, size_t argc, char *argv[])
 {
 	int err;
 
-	err = cmd_config(sh, argc, argv);
-	if (err) {
-		return err;
+	if (default_stream == NULL) {
+		shell_print(sh, "Not connected");
```
Comment (MariuszSkamra) [inline subsys/bluetooth/shell/audio.c:1201]:
```suggestion
		shell_error(sh, "No stream selected");
```

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]
Minor issue to fix, otherwise LGTM

**Review by szymon-czapracki — DISMISSED** [review]

**Review by MariuszSkamra — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Review by asbjornsabo — APPROVED** [review]
LGTM

**Pull request merged**